### PR TITLE
Avoid runtime errors when requested resource doesn't exist

### PR DESCRIPTION
**Pull request by spadgett**
Update DataService to tolerate requests for resources that were not
found during discovery. This will show an error notification and reject
the promise rather than resulting in a runtime error.

/assign @jwforres 
@deads2k FYI

### CODE REVIEW

File: dist/origin-web-common-services.js [hunk 1]
```diff
@@ -1521,6 +1521,27 @@ angular.module('openshiftCommonServices')
     addQueuedNotifications();
   };
 
+  var unknownResourceError = function(resource, opts) {
+    var message;
+    if (!resource) {
+      message = 'Interal error: API resource not specified.';
```
Comment (jwforres) [inline dist/origin-web-common-services.js]: typo, Internal

**Comment by jwforres** [comment]
so this covers all the XHR, what are you thinking for the websockets?

**Comment by spadgett** [comment]
We list before we open the websocket, so the list will fail.

**Comment by jwforres** [comment]
@spadgett what about log streaming

**Comment by jwforres** [comment]
i guess we get the pod first

**Comment by spadgett** [comment]
Updated `DataService.createStream()`

**Comment by spadgett** [comment]
Rebased to pick up #303

**Comment by jwforres** [comment]
sorry just noticed the change to uniqueKey, how is that going to impact the functions relying on it, are any of them going to blow up with a null key

**Pull request merged**